### PR TITLE
feat: add color system and color docs

### PR DESCRIPTION
**Pull request by abutterworth**
This PR includes a color system built on top of bootstrap's existing color system. The notion of color levels is expanded and affords setting explicit values for all theme color levels rather than bootstrap's use of `darken()` or `lighten()`.

Accessibility: All color levels 500 and greater are intended to provide accessible contrast on white and color levels 100 and lower.

Color utilities have been added with the intention that Open edX users should be able to apply a bootstrap theme variables file to Paragon SCSS without breaking the output. This includes making sure our additions have reasonable defaults if custom variables are not provided.

Button and Alert SCSS definitions have been replaced wholesale by Paragon in this PR. This provides us greater control over colors and focus states.

Screen Shot 2019-07-02 at 1 14 42 PM

### CODE REVIEW

File: www/src/scss/_code-block.scss [hunk 1]
```diff
@@ -9,5 +9,5 @@
 }
 .pgn-doc__code-block-error {
   padding: .5rem;
-  background: $fill-danger;
+  // background: $fill-danger;
```
Comment (davidjoy) [inline www/src/scss/_code-block.scss]: This comment left in intentionally?

Comment (abutterworth) [reply www/src/scss/_code-block.scss]: Nope, thanks for catching

File: scss/core/_functions.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
+//
+// BOOTSTRAP FUNCTION OVERRIDE
+// Request a theme color. Now with a variant option
```
Comment (albemarle) [inline scss/core/_functions.scss]: Period after "option"?

File: scss/core/extensions/_utilities.scss [hunk 3]
```diff
@@ -1,13 +1,12 @@
-@each $color, $value in $colors {
-  @include bg-variant(".bg-#{$color}", $value);
-}
-
-@each $color, $value in $colors {
-  @include text-emphasis-variant(".text-#{$color}", $value);
-}
+// Add background, border and text color utilities
```
Comment (albemarle) [inline scss/core/extensions/_utilities.scss]: How do you feel about the Oxford comma?  _ducking & running_

File: scss/core/mixins/_buttons.scss [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    }
+
+    &:active,
+    &.active {
+      &:before {
+        opacity: .75;
+      }
+    }
+
+    // Disabled buttons should not have focus rings
```
Comment (albemarle) [inline scss/core/mixins/_buttons.scss]: Nice comments.  👍

File: scss/core/overrides/_alert.scss [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    right: 0;
+    padding: $alert-padding-y $alert-padding-x;
+    color: inherit;
+  }
+}
+
+
+// Alternate styles
+//
+// Generate contextual modifier classes for colorizing the alert.
```
Comment (albemarle) [inline scss/core/overrides/_alert.scss]: "coloring" ?

File: scss/core/variables/_colors.scss [hunk 6]
```diff
@@ -0,0 +1,122 @@
+
+// Color system
+
+$white:    #fff !default;
+$gray-100: #F9F9F9 !default;
+$gray-200: #b9babe !default;
+$gray-300: #A2A4A9 !default;
+// $gray-400: #A2A2A2 !default; // TODO: do we need this gray?
```
Comment (albemarle) [inline scss/core/variables/_colors.scss]: Just checking if this is left in on purpose.

File: scss/edx/variables/_colors.scss [hunk 7]
```diff
@@ -0,0 +1,122 @@
+
+// Color system
+
+$white:    #fff !default;
+$gray-100: #F9F9F9 !default;
+$gray-200: #b9babe !default;
+$gray-300: #A2A4A9 !default;
+// $gray-400: #A2A2A2 !default; // TODO: do we need this gray?
```
Comment (albemarle) [inline scss/edx/variables/_colors.scss]: Same question here.

File: www/src/pages/foundations/colors.jsx [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+];
+
+const selectorColors = {};
+
+function parseColors(cssSelectors) {
+  const colorsAreParsed = Object.keys(selectorColors).length !== 0;
+  if (colorsAreParsed) return;
+
+  cssSelectors.forEach(({ selector, declarations }) => {
+    // All delcarations fit this shape: "background-color: #fff !important;"
```
Comment (albemarle) [inline www/src/pages/foundations/colors.jsx]: "declarations"

File: www/src/pages/foundations/colors.jsx [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+    const declarationFragments = declarations[0].split(' ');
```
Comment (albemarle) [inline www/src/pages/foundations/colors.jsx]: Seems like this and the right-hand side of the assignment below don't need to be in the forEach?

File: www/src/pages/foundations/colors.jsx [hunk 10]
```diff
@@ -0,0 +1,320 @@
...
+  parseColors(data.allCssUtilityClasses.nodes);
+
+  return (
+    <div>
+
+      <SEO title="Colors" />
+
+      <h1>Colors</h1>
+
+      <h3>Pallete</h3>
```
Comment (albemarle) [inline www/src/pages/foundations/colors.jsx]: "Palette"

File: scss/core/variables/_colors.scss (hunk 6 above)
Comment (abutterworth) [reply scss/core/variables/_colors.scss]: I did leave them there on purpose, but I suppose easy enough to add back in if needed. Will delete.

File: scss/core/extensions/_utilities.scss (hunk 3 above)
Comment (abutterworth) [reply scss/core/extensions/_utilities.scss]: ducking, running, & yelling! ;)

**Pull request merged**